ui/cap-react/src/antd/forms/widgets/SelectWidget.js

```diff
@@ -0,0 +1,179 @@
+import { useState } from "react";
+import { connect } from "react-redux";
+import Select from "antd/lib/select";
+import { debounce } from "lodash-es";
+import axios from "../../../axios";
+import { fromJS } from "immutable";
+import { Empty } from "antd";
+import PropTypes from "prop-types";
+
+import {
+  ariaDescribedByIds,
+  enumOptionsIndexForValue,
+  enumOptionsValueForIndex,
+} from "@rjsf/utils";
+import isString from "lodash-es";
+
+const SelectWidget = ({
+  autofocus,
+  disabled,
+  formContext,
+  id,
+  multiple,
+  onBlur,
+  onChange,
+  onFocus,
+  options,
+  placeholder,
+  readonly,
+  value,
+  formData,
+}) => {
+  const { readonlyAsDisabled = true } = formContext;
+
+  const { enumOptions, enumDisabled, suggestions, params, emptyValue } =
+    options;
+
+  const [loading, setLoading] = useState(false);
+  const [data, setData] = useState([]);
+
+  const handleChange = nextValue => {
+    onChange(
+      enumOptionsValueForIndex(nextValue, enumOptions || data, emptyValue)
+    );
+  };
+
+  const handleBlur = () =>
+    onBlur(id, enumOptionsValueForIndex(value, enumOptions, emptyValue));
+
+  const handleFocus = () =>
+    onFocus(id, enumOptionsValueForIndex(value, enumOptions, emptyValue));
+
+  const handleSearch = newValue => {
+    updateSearch(newValue, setData);
+  };
+
+  const filterOption = (input, option) => {
+    if (option && isString(option.label)) {
+      // labels are strings in this context
+      return option.label.toLowerCase().indexOf(input.toLowerCase()) >= 0;
+    }
+    return false;
+  };
+
+  const getPopupContainer = node => node.parentNode;
+
+  console.log(value, enumOptions, multiple);
+
+  const selectedIndexes = enumOptionsIndexForValue(
+    value,
+    enumOptions,
+    multiple
+  );
+
+  const _replace_hash_with_current_indexes = path => {
+    let indexes = id.split("_").filter(item => !isNaN(item)),
+      index_cnt = 0;
+
+    return path.map(item => {
+      item = item === "#" ? indexes[index_cnt] : item;
+      if (!isNaN(item)) ++index_cnt;
+      return item;
+    });
+  };
+
+  const updateSearch = (value, callback) => {
+    let data = fromJS(formData);
+    if (params) {
```

Comment:
I'm thinking that this if block and the function above could be removed. I don't know if there's a use case for it @pamfilos

Reply:
There's in fact a use case in cms.json > primary_datasets > triggers, so we keep it